### PR TITLE
Initial commit for #279.

### PR DESCRIPTION
**Pull request by cynthia**
There needs to be some extra details fleshed out.

***
<a href="https://pr-preview.s3.amazonaws.com/w3ctag/design-principles/pull/381.html" title="Last updated on Jul 26, 2022, 10:59 AM UTC (253803d)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3ctag/design-principles/381/eaf5072...253803d.html" title="Last updated on Jul 26, 2022, 10:59 AM UTC (253803d)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -994,6 +994,22 @@ Consider adding such features only in cases when the overall user experience is
...
+<h3 id="html-idl-must-by-synced">Keep attributes in sync</h3>
+
+If your feature introduces new attributes,
+try to ensure the attributes are synchronized.
+
+This means any new content attribute
+should always have a corresponding IDL attribute with the same name,
+and the state between the two should be kept synchronized.
+Carving out a synchronized IDL attribute with inconsistent naming
+results in confusion, and should be avoided at all costs.
```
Comment (hober) [inline index.bs:1006]: "at all costs" is overstating it, isn't it?

File: index.bs [hunk 2]
```diff
@@ -994,6 +994,22 @@ Consider adding such features only in cases when the overall user experience is
 A canonical example of this is blocking rendering in order to download and process a stylesheet. 
 The alternative user experience is a flash of unstyled content, which is undesirable.
 
+<h3 id="html-idl-must-by-synced">Keep attributes in sync</h3>
+
+If your feature introduces new attributes,
+try to ensure the attributes are synchronized.
```
Comment (LeaVerou) [inline index.bs:1000]: This doesn't quite make sense. The attributes are synchronized with what? With each other?

File: index.bs [hunk 3]
```diff
@@ -994,6 +994,22 @@ Consider adding such features only in cases when the overall user experience is
...
+
+This means any new content attribute
```
Comment (hober) [inline index.bs:1002]:
```suggestion
This means new content attributes
```

File: index.bs [hunk 4]
```diff
@@ -994,6 +994,22 @@ Consider adding such features only in cases when the overall user experience is
...
+should always have a corresponding IDL attribute with the same name,
```
Comment (hober) [inline index.bs:1003]:
```suggestion
should have a corresponding IDL attribute with the same name,
```

File: index.bs (hunk 2 above)
Comment (LeaVerou) [inline index.bs:1000]: How about this?
```suggestion
If your feature introduces new HTML attributes,
it should also introduce corresponding IDL attributes,
and these should be synchronized.
```

File: index.bs (hunk 1 above)
Comment (hober) [inline index.bs:1006]:
```suggestion
results in confusion, and should be avoided.
```

**Review by hober — APPROVED** [review]
Looks good! Here are some nits.

File: index.bs [hunk 5]
```diff
@@ -994,6 +994,22 @@ Consider adding such features only in cases when the overall user experience is
...
+and the state between the two should be kept synchronized.
+Carving out a synchronized IDL attribute with inconsistent naming
+results in confusion, and should be avoided at all costs.
+
+<div class="note">
+This does not hold the other way around.
+A new IDL attribute does not always warrant a content attribute counterpart.
```
Comment (LeaVerou) [inline index.bs:1010]: This needs an example of an IDL attribute that does not correspond to an HTML attribute.

File: index.bs (hunk 1 above)
[the remark this answers is not in the document]
Comment (LeaVerou) [reply index.bs:1007]: And we should definitely mention the classic examples of `<input value>` / `<input checked>` somewhere in this.

File: index.bs (hunk 4 above)
Comment (LeaVerou) [reply index.bs]: Now that you merged my suggestion, we are basically saying the same thing twice. We should merge paragraphs 1 and 2.

File: index.bs [hunk 6]
```diff
@@ -1001,6 +1001,23 @@ Consider adding such features only in cases when the overall user experience is
...
 
+<h3 id="html-idl-must-by-synced">Keep attributes in sync</h3>
+
+If your feature introduces new HTML attributes,
+it should also introduce corresponding IDL attributes,
+and these should be synchronized.
+
+This means new content attributes
+should have a corresponding IDL attribute with the same name,
+and the state between the two should be kept synchronized.
```
Comment (LeaVerou) [inline index.bs:1012]:
```suggestion
New content attributes
should have a corresponding IDL attribute with the same name,
and the state between the two should be kept synchronized.
```

File: index.bs (hunk 1 above)
[the remark this answers is not in the document]
Comment (cynthia) [reply index.bs]: See: https://github.com/w3ctag/design-principles/pull/381/commits/357a8da48be7e7456732db8150bd3df776391db2

File: index.bs [hunk 7]
```diff
@@ -1001,6 +1001,23 @@ Consider adding such features only in cases when the overall user experience is
...
+Carving out a synchronized IDL attribute with inconsistent naming
+results in confusion, and should be avoided.
```
Comment (LeaVerou) [inline index.bs:1010]:
```suggestion
results in confusion, and should be avoided.

<div class=example>

The <{input}> element has <{input/value}> and <{input/checked}> attributes that hold the default state of the element. 
There are also {{HTMLInputElement/value}} and  {{HTMLInputElement/checked}} that hold the *current* state of the element.
The fact that these are not synchronized, and despite the same name, they actually hold different data, has caused authors a lot of confusion. 

</div>
```

**Review by ylafon — APPROVED** [review]
LGTM

**Pull request merged**